lib/stream.js

```diff
@@ -83,20 +95,30 @@ const resumeStream = async stream => {
 
 const applyEncoding = (contents, encoding) => encoding === 'buffer' ? new Uint8Array(contents) : contents;
 
+// Handles `childProcess.stdin` aborting or erroring.
+// But ignores the `childProcess.stdin.destroy()` automatically done on child process exit by Node.js.
+// `controller.signal` is used to ensure this never resolves, only rejects.
+const throwOnInputStdioError = ({spawned, stdioStreamsGroups, processExitPromise, controller}) => spawned.stdio
+	.filter((stream, index) => stream !== null && stdioStreamsGroups[index][0].direction === 'input')
+	.map(stream => throwOnInputStreamError(stream, processExitPromise, controller));
+
+const throwOnInputStreamError = async (stream, processExitPromise, {signal}) => {
+	await Promise.all([
+		Promise.race([
+			processExitPromise,
```

Comment:
This is due to:

https://github.com/nodejs/node/blob/eca4b1a403959d5b83fb5a9adaae8f4c3580eea2/lib/internal/child_process.js#L278

Calling `childProcess.stdin.destroy()` while a child process is running should be handled as abnormal, since that's aborting its input.
However, Node.js calls `childProcess.stdin.destroy()` when the child process ends, and that is normal. 
Therefore, we use `Promise.race()` so that `finished()` does nothing once the child process has ended.

Reply:
Maybe add this as a short code comment.

Reply:
Fixed :+1: